src/AstTransforms.cpp

```diff
@@ -221,6 +222,146 @@ bool UniqueAggregationVariablesTransformer::transform(AstTranslationUnit& transl
     return changed;
 }
 
+bool MaterializeSingletonAggregationTransformer::transform(
+				AstTranslationUnit& translationUnit) {
+		bool changed = false;
+		AstProgram& program = *translationUnit.getProgram();
+    const TypeEnvironment& env = translationUnit.getAnalysis<TypeEnvironmentAnalysis>()->getTypeEnvironment();
+		int counter = 0;
+		visitDepthFirst(program, [&](const AstClause& clause) {
+			visitDepthFirst(clause, [&](const AstLiteral& literal) {
+				visitDepthFirst(literal, [&](const AstAggregator& agg) {
+					if (!isSingleValued(agg)) {
+							return;
+					}
+					changed = true; 
+					auto* clause = new AstClause();
+					auto* head = new AstAtom();
+					head->setQualifiedName("A");
+					head->addArgument(std::make_unique<AstVariable>("X"));
+					clause->setHead(std::unique_ptr<AstAtom>(head));
+					// auto* literalCopy = literal.clone();
+					clause->addToBody(std::unique_ptr<AstLiteral>(head->clone()));
+					program.addClause(std::unique_ptr<AstClause>(clause));
+//					// We are now in this situation
```

Comment:
This is an example of where I add the rule 
A(X) :- A(X).
To a program. (I run souffle on a program that has a relation A already declared)
If I make the body out of the cloned literal instead of the cloned head, I segfault. I can't reason about why. I need help from you guys to understand why this would happen.